types/src/transaction.rs

```diff
@@ -1210,3 +1211,22 @@ impl IntoProto for TransactionListWithProof {
         out
     }
 }
+
+/// `ExecutorTransaction` will be the transaction type used internally in the libra node to
+/// represent the transaction to be processed and persisted.
+///
+/// We suppress the clippy warning here as we would expect most of the transaction to be user
+/// transaction.
+#[allow(clippy::large_enum_variant)]
+pub enum ExecutorTransaction {
```

Comment:
I just feel `ExecutorTransaction` is a bit weird name. The concept is not completely internal because clients will need to interact with it (if they request a transaction with proof, this is what they get). I don't have great suggestion in mind right now, but maybe just `Transaction` is good.

Reply:
Seems reasonable to rename current `Transaction` concept to `UserTransaction` (hence `UserTransactionPayload`, `SignedUserTransaction`, etc.).  cc @sunshowers

Reply:
Yea I struggled to find good name for it. The main reason is that we have this notion of `SignedTransaction` right now. I do like the name of `Transaction` vs `UserTransaction` but IMHO we should do that in a separate PR.

Reply:
I think `Transaction` is good because this includes all kinds of transactions we have in the system.